paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/PaymentSheetPlaygroundViewModel.kt

```diff
@@ -177,7 +311,7 @@ internal class PaymentSheetPlaygroundViewModel(
         paymentMethod: PaymentMethod,
         shouldSavePaymentMethod: Boolean,
     ): CreateIntentResult {
-        val playgroundState = state.value ?: return CreateIntentResult.Failure(
+        val playgroundState = state.value as? PlaygroundState.Payment ?: return CreateIntentResult.Failure(
```

Comment:
We're doing this a ton, I think it definitely makes sense to have an extension function for this.